src/index/CompressedBlockPrefiltering.h

```diff
@@ -0,0 +1,140 @@
+  // Helper function that implements the relational comparison on the block
+  // values.
+  bool compareImpl(const ValueId& tripleId, const ValueId& otherId) const;
+  // Helper for providing the lower indices during the evaluation procedure.
+  auto lowerIndex(std::vector<BlockMetadata>& input, size_t evaluationColumn);
+  // Helper to get the upper indices, necessary for EQ and NE.
+  auto upperIndex(std::vector<BlockMetadata>& input, size_t evaluationColumn)
+      requires(Comparison == CompOp::EQ || Comparison == CompOp::NE);
+};
+
+//______________________________________________________________________________
+// Helper struct for a compact class implementation regarding the logical
+// operations `AND`, `OR` and `NOT`.
+enum struct LogicalOperators { AND, OR, NOT };
+
+//______________________________________________________________________________
+template <LogicalOperators Operation>
+class LogicalExpressions : public PrefilterExpression {
+ public:
+  explicit LogicalExpressions(std::unique_ptr<PrefilterExpression> child1,
+                              std::unique_ptr<PrefilterExpression> child2)
+      : child1_(std::move(child1)), child2_(std::move(child2)) {}
+
+  std::vector<BlockMetadata> evaluate(std::vector<BlockMetadata>& input,
+                                      size_t evaluationColumn) override;
+
+ private:
+  std::unique_ptr<PrefilterExpression> child1_;
+  std::unique_ptr<PrefilterExpression> child2_;
+  // Introduce a helper method for the specification LogicalOperators::OR
+  std::vector<BlockMetadata> evaluateOr(std::vector<BlockMetadata>& input,
+                                        size_t evaluationColumn)
+      requires(Operation == LogicalOperators::OR);
+};
+
+}  // namespace prefilterExpressions
+
+//______________________________________________________________________________
+// Definition of the RelationalExpressions for LT, LE, EQ, NE, GE and GT.
+using LessThanExpression = prefilterExpressions::RelationalExpressions<
+    prefilterExpressions::CompOp::LT>;
+using LessEqualExpression = prefilterExpressions::RelationalExpressions<
+    prefilterExpressions::CompOp::LE>;
+using EqualExpression = prefilterExpressions::RelationalExpressions<
+    prefilterExpressions::CompOp::EQ>;
+using NotEqualExpression = prefilterExpressions::RelationalExpressions<
+    prefilterExpressions::CompOp::NE>;
+using GreaterEqualExpression = prefilterExpressions::RelationalExpressions<
+    prefilterExpressions::CompOp::GE>;
+using GreaterThanExpression = prefilterExpressions::RelationalExpressions<
+    prefilterExpressions::CompOp::GT>;
+
+//______________________________________________________________________________
+// Definition of the LogicalExpressions for AND and OR.
+using AndExpression = prefilterExpressions::LogicalExpressions<
+    prefilterExpressions::LogicalOperators::AND>;
+using OrExpression = prefilterExpressions::LogicalExpressions<
+    prefilterExpressions::LogicalOperators::OR>;
```

Comment:
I think those aliases should also be inside the namespace, s.t. the name is
`prefilterExpressions::OrExpression` to nicely disambiguate with the "ordianary" expressions.